lib/src/Recorder.cpp

```diff
@@ -1041,14 +1043,18 @@ void Recorder::recordQueryAattributeEnumValuesCapability(
         return;
     }
 
-    auto key = sai_serialize_object_type(SAI_OBJECT_TYPE_SWITCH) + ":" + sai_serialize_object_id(switchId);
+    if (!meta->enummetadata)
+    {
+        SWSS_LOG_ERROR("Attribute %s is not enum/enumlist!", meta->attridname);
+        return;
+    }
 
-    sai_attribute_t attr;
+    auto key = sai_serialize_object_type(SAI_OBJECT_TYPE_SWITCH) + ":" + sai_serialize_object_id(switchId);
 
-    attr.id = attrId;
-    attr.value.s32list = *enumValuesCapability;
+    auto str_attr_id = sai_serialize_attr_id(*meta);
+    auto str_enum_list = sai_serialize_enum_list(*enumValuesCapability, meta->enummetadata, true); // we only need to serialize count
```

Comment:
also, does this problem applies to RedisRemoteInterfaceSai.cpp ? and same on Syncd.cpp on desterilize and need to be corrected in both of them?

Reply:
No, in RedisRemoteInterfaceSai/Syncd there is different serialization/deserialization implemented. All works well, it just the recorder that fails to record when querying SAI_DEBUG_COUNTER_ATTR_TYPE